presto-main/src/main/java/com/facebook/presto/operator/OperationTimer.java

```diff
@@ -53,11 +60,18 @@
         this.trackOperationCpuTime = trackOperationCpuTime;
         checkArgument(trackOverallCpuTime || !trackOperationCpuTime, "tracking operation cpu time without tracking overall cpu time is not supported");
 
+        this.trackOverallAllocation = trackOverallAllocation;
+        this.trackOperationAllocation = trackOperationAllocation;
+        checkArgument(trackOverallAllocation || !trackOperationAllocation, "tracking operation allocation without tracking overall allocation is not supported");
+        checkArgument(!trackOverallAllocation || THREAD_MX_BEAN_EX != null, "tracking allocation is not supported by this JVM");
```

Comment:
- This is simpler to understand  :) 
     `!(trackOverallAllocation && THREAD_MX_BEAN_EX == null)`  
    `!(trackOperationAllocation && !trackOverallAllocation)` 
- Move second check up

Reply:
@viczhang861 Changed to

```
        this.trackOverallAllocation = trackOverallAllocation;
        this.trackOperationAllocation = trackOperationAllocation;
        if (trackOverallAllocation) {
            checkArgument(THREAD_MX_BEAN_EX != null, "tracking allocation is not supported by this JVM");
        }
        else {
            checkArgument(!trackOperationCpuTime, "tracking operation cpu time without tracking overall cpu time is not supported");
        }
```

Reply:
Much better. I restarted a failed test.